pkg/pipeline/extract/aggregate/aggregate.go

```diff
@@ -47,7 +46,7 @@ type NormalizedValues string
 type Aggregate struct {
 	Definition api.AggregateDefinition
 	GroupsMap  map[NormalizedValues]*GroupState
-	GroupsList *list.List
+	Cache      *utils.TimedCache
```

Comment:
This change, replaced a linked list with a `TimedCache` (which is a linked list and a map). Can we remove `GroupsMap` and use the new cache instead?

Reply:
The GroupsMap is used in several places in aggregate.go, and I think it would be messy to use the cache instead. If you see a clean way to do it, please share.

Reply:
I found 2 uses of `GroupsMap` in `aggregate.go`:
1. https://github.com/netobserv/flowlogs-pipeline/blob/9783cf63f9545c32e1124f48a1e4faa74be5438e/pkg/pipeline/extract/aggregate/aggregate.go#L138
I think it can be replaced with a call to `UpdateCacheEntry()`

2. https://github.com/netobserv/flowlogs-pipeline/blob/9783cf63f9545c32e1124f48a1e4faa74be5438e/pkg/pipeline/extract/aggregate/aggregate.go#L208
We need to add a new method to support iterating over the items in the cache.
I suggest adding a method like this:
```
func (l *TimedCache) Iterate(f func(key string, value interface{})) {
	for k, v := range l.CacheMap {
		f(k, v.SourceEntry)
	}
}
```

and replace the for loop with something like:
```
aggregate.Cache.Iterate(func(key string, value interface{}) {
	group := value.(GroupState)
	metrics = append(metrics, config.GenericMap{
	....
})
```